sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADJwtBearerTokenAuthenticationConverter.java

```diff
@@ -2,74 +2,70 @@
 // Licensed under the MIT License.
 package com.azure.spring.aad.webapi;
 
+import com.azure.spring.aad.AADOAuth2AuthenticatedPrincipal;
+import com.azure.spring.aad.AbstractJwtBearerTokenAuthenticationConverter;
 import org.springframework.core.convert.converter.Converter;
-import org.springframework.security.authentication.AbstractAuthenticationToken;
 import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.OAuth2AuthenticatedPrincipal;
 import org.springframework.security.oauth2.jwt.Jwt;
 import org.springframework.security.oauth2.server.resource.authentication.BearerTokenAuthentication;
-import org.springframework.security.oauth2.server.resource.authentication.JwtGrantedAuthoritiesConverter;
-import org.springframework.util.Assert;
 
 import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.azure.spring.aad.AADJwtGrantedAuthoritiesConverter.DEFAULT_AUTHORITY_PREFIX;
+import static com.azure.spring.aad.AADJwtGrantedAuthoritiesConverter.DEFAULT_CLAIM_TO_AUTHORITY_PREFIX_MAP;
 
 /**
  * A {@link Converter} that takes a {@link Jwt} and converts it into a {@link BearerTokenAuthentication}.
  */
-public class AADJwtBearerTokenAuthenticationConverter implements Converter<Jwt, AbstractAuthenticationToken> {
-
-    private static final String DEFAULT_AUTHORITY_PREFIX = "SCOPE_";
-    private static final String DEFAULT_PRINCIPAL_CLAIM_NAME = "sub";
-
-    private Converter<Jwt, Collection<GrantedAuthority>> jwtGrantedAuthoritiesConverter;
-    private String principalClaimName = DEFAULT_PRINCIPAL_CLAIM_NAME;
+public class AADJwtBearerTokenAuthenticationConverter extends AbstractJwtBearerTokenAuthenticationConverter {
 
     /**
-     * Use AADJwtGrantedAuthoritiesConverter, it can resolve the access token of scp and roles.
+     * Construct AADJwtBearerTokenAuthenticationConverter by DEFAULT_PRINCIPAL_CLAIM_NAME and DEFAULT_CLAIM_TO_AUTHORITY_PREFIX_MAP.
      */
     public AADJwtBearerTokenAuthenticationConverter() {
-        this.jwtGrantedAuthoritiesConverter = new AADJwtGrantedAuthoritiesConverter();
+        this(DEFAULT_PRINCIPAL_CLAIM_NAME, DEFAULT_CLAIM_TO_AUTHORITY_PREFIX_MAP);
     }
 
     /**
-     * Using spring security provides JwtGrantedAuthoritiesConverter, it can resolve the access token of scp or roles.
-     *
-     * @param authoritiesClaimName authorities claim name
+     * Construct AADJwtBearerTokenAuthenticationConverter with the authority claim.
+     * @param authoritiesClaimName authority claim name
      */
     public AADJwtBearerTokenAuthenticationConverter(String authoritiesClaimName) {
         this(authoritiesClaimName, DEFAULT_AUTHORITY_PREFIX);
```

Comment:
nit: this(null, buildClaimToAuthorityPrefixMap(authoritiesClaimName, DEFAULT_AUTHORITY_PREFIX));